pkg/tasks/configsharing.go

```diff
@@ -71,7 +74,27 @@ func (t *ConfigSharingTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "failed to retrieve Thanos Querier host")
 	}
 
-	cm := t.factory.SharingConfig(promURL, amURL, thanosURL)
+	var svc *v1.Service
+	if t.config.UserWorkloadConfiguration.Alertmanager.Enabled {
+		svc, err = t.factory.AlertmanagerUserWorkloadService()
+		if err != nil {
+			return errors.Wrap(err, "initializing Alertmanager User Workload Service failed")
+		}
+	} else {
+		svc, err = t.factory.AlertmanagerService()
+		if err != nil {
+			return errors.Wrap(err, "initializing Alertmanager Service failed")
+		}
+	}
```

Comment:
I can't figure out why we use the Alertmanager Service if UWM alertmanger is disabled. Might be worth leaving a comment for similarly dense future developers, i.e. me in 6 month..?

Reply:
good call, I've also realized that the port numbers could vary so I've updated the patch anyway.